### DISCUSSION TITLE
API Key Management

### DISCUSSION THREAD

**Discussion by DrakezulsMinimalism** [DependencyTrack/dependency-track#1010] [Ideas]
Hi everyone,
I'm trying to introduce DependencyTrack into my organization and added a few users of different development projects.
Naturally, I do not want to give all of them user access management permissions, but how can they get an API key then?

Maybe they could either "take" an existing API key of their assigned teams (if one exists) or maybe individual users could generate named API keys that are additionally bound to a specific team?
Binding to user and the user's team at the same time is to limit project access, if team->project relationships are eventually introduced.

**Comment by 37b** [comment]
I would also love to have users manage their own API keys. Though as an Admin I would want the ability to destroy keys at will.
